OData/src/System.Web.OData/OData/Query/Expressions/ExpressionBinderBase.cs

```diff
@@ -511,6 +554,154 @@ internal Expression BindHas(Expression left, Expression flag)
+            MemberInitExpression nextExpression = null;
+            Expression nestedExpression = null;
+            foreach (var expr in expression.Bindings.OfType<MemberAssignment>())
+            {
+                var initExpr = expr.Expression as MemberInitExpression;
+                if (initExpr != null && expr.Member.Name == "Next")
+                {
+                    nextExpression = initExpr;
+                }
+                else if (expr.Member.Name == "Name")
+                {
+                    nameToAdd = (expr.Expression as ConstantExpression).Value as string;
+                }
+                else if (expr.Member.Name == "Value" || expr.Member.Name == "NestedValue")
+                {
+                    resultType = expr.Expression.Type;
+                    if (resultType == typeof(object) && expr.Expression.NodeType == ExpressionType.Convert)
+                    {
+                        resultType = ((UnaryExpression)expr.Expression).Operand.Type;
+                    }
+
+                    if (typeof(GroupByWrapper).IsAssignableFrom(resultType))
+                    {
+                        nestedExpression = expr.Expression;
+                    }
+                }
+            }
+
+            if (prefix != null)
+            {
+                nameToAdd = prefix + "\\" + nameToAdd;
+            }
+
+            if (typeof(GroupByWrapper).IsAssignableFrom(resultType))
+            {
+                flattenPropertyContainer.Add(nameToAdd, Expression.Property(source, "NestedValue"));
+            }
+            else
+            {
+                flattenPropertyContainer.Add(nameToAdd, Expression.Convert(Expression.Property(source, "Value"), resultType));
+            }
+
+            if (nextExpression != null)
+            {
+                CollectAssigments(flattenPropertyContainer, Expression.Property(source, "Next"), nextExpression, prefix);
+            }
+
+            if (nestedExpression != null)
+            {
+                var nestedAccessor = ((nestedExpression as MemberInitExpression).Bindings.First() as MemberAssignment).Expression as MemberInitExpression;
+                var newSource = Expression.Property(Expression.Property(source, "NestedValue"), "GroupByContainer");
+                CollectAssigments(flattenPropertyContainer, newSource, nestedAccessor, nameToAdd);
+            }
+        }
+
+        /// <summary>
+        /// Gets expression for property from previously aggregated query
+        /// </summary>
+        /// <param name="propertyPath"></param>
+        /// <returns>Returns null if no aggregations were used so far</returns>
+        protected Expression GetFlattenedPropertyExpression(string propertyPath)
+        {
+            if (flattenedPropertyContainer == null)
+            {
+                return null;
+            }
+
+            Expression expression;
+            if (flattenedPropertyContainer.TryGetValue(propertyPath, out expression))
```

Comment:
The lines 696-700 should be replaced by the following lines to allow ODataConventionModelBuilderExtensions.EnableLowerCamelCase() to work. See issue #1136

```
#            var entry = FlattenedPropertyContainer.FirstOrDefault(c => string.Equals(c.Key, propertyPath, StringComparison.OrdinalIgnoreCase));
            if (entry.Key != null)
            {
                return entry.Value;
            }
```